services/wallet/reader.go

```diff
@@ -479,13 +479,13 @@ func (r *Reader) GetWalletToken(ctx context.Context, clients map[uint64]chain.Cl
 
 	var (
 		group             = async.NewAtomicGroup(ctx)
-		prices            = map[string]map[string]float64{}
+		prices            = map[string]map[string]market.DataPoint{}
 		tokenDetails      = map[string]thirdparty.TokenDetails{}
 		tokenMarketValues = map[string]thirdparty.TokenMarketValues{}
 	)
 
 	group.Add(func(parent context.Context) error {
-		prices, err = r.marketManager.FetchPrices(tokenSymbols, currencies)
+		prices, err = r.marketManager.GetOrFetchPrices(tokenSymbols, currencies, 60)
```

Comment:
Here we're changing the way `GetWalletToken` will request token price data. Now token price data will be cached for `60` seconds when calling `GetWalletToken`.